.changeset/petite-needles-act.md

```
@@ -0,0 +1,5 @@
+---
+'@mastra/core': patch
+---
+
+Fixed manual thread renames being overwritten by Observational Memory. Renames now pin the title by default; pass pin: false for initial titles or regenerate the title to resume automatic naming.
```

markdownlint-cli2 (0.23.2)

[warning] First line in a file should be a top-level heading

(MD041, first-line-heading, first-line-h1)